File: lib/ParticleFilter/Map/door.cpp
#include "door.h"

void Door::from_json(const json &j, Door &doorData)
{
    j.at("id").get_to(doorData.id); 
    j.at("startX").get_to(doorData.startX);
    j.at("startY").get_to(doorData.startY);
    j.at("stopX").get_to(doorData.stopX);
    j.at("stopY").get_to(doorData.stopY);
}

int Door::getWidth() {
    return stopX - startX;
}

int Door::getHeight() {
    return stopY - startY;
}

/// @brief Check if the given x/y coordinate is inside the door.
/// @param x X coordinate to check.
/// @param y Y coordinate to check.
/// @return Whether or not the coordinate is inside the door.
bool Door::containsPoint(const int x, const int y) const
{
    return x >= startX && x <= stopX && y >= startY && y <= stopY;
}

File: Python/LocalizationTable/Map/Door.py
from .rectangle import Rectangle


class Door(Rectangle):
    def __init__(self, id: int, start_x: int, stop_x: int, start_y: int, stop_y: int):
        super().__init__(id, start_x, stop_x, start_y, stop_y)

    @staticmethod
    def from_json(json_data):
        return Door(json_data["id"],
                    json_data["startX"],
                    json_data["stopX"],
                    json_data["startY"],
                    json_data["stopY"])
